train_segmentation.py

```diff
@@ -487,59 +504,6 @@ def main(params, config_path):
-    print(f"Number of cuda devices requested: {params['global']['num_gpus']}. Cuda devices available: {lst_device_ids}\n")
-    if num_devices == 1:
-        print(f"Using Cuda device {lst_device_ids[0]}\n")
-    elif num_devices > 1:
-        # TODO: why are we showing indices [1:-1] for lst_device_ids?
-        print(f"Using data parallel on devices: {str(lst_device_ids)[1:-1]}. Main device: {lst_device_ids[0]}\n")
-        try:  # For HPC when device 0 not available. Error: Invalid device id (in torch/cuda/__init__.py).
-            model = nn.DataParallel(model, device_ids=lst_device_ids)  # DataParallel adds prefix 'module.' to state_dict keys
-        except AssertionError:
-            warnings.warn(f"Unable to use devices {lst_device_ids}. Trying devices {list(range(len(lst_device_ids)))}")
-            device = torch.device('cuda:0')
-            lst_device_ids = range(len(lst_device_ids))
-            model = nn.DataParallel(model,
-                                    device_ids=lst_device_ids)  # DataParallel adds prefix 'module.' to state_dict keys
-
-    else:
-        warnings.warn(f"No Cuda device available. This process will only run on CPU\n")
-
-    dontcare = get_key_def("ignore_index", params["training"], -1)
-    if dontcare == 0:
-        warnings.warn("The 'dontcare' value (or 'ignore_index') used in the loss function cannot be zero;"
-                      " all valid class indices should be consecutive, and start at 0. The 'dontcare' value"
-                      " will be remapped to -1 while loading the dataset, and inside the config from now on.")
-        params["training"]["ignore_index"] = -1
-
-    tqdm.write(f'Creating dataloaders from data in {samples_folder}...\n')
-    trn_dataloader, val_dataloader, tst_dataloader = create_dataloader(samples_folder=samples_folder,
-                                                                       batch_size=batch_size,
-                                                                       num_devices=num_devices,
-                                                                       params=params)
-
-    tqdm.write(f'Setting model, criterion, optimizer and learning rate scheduler...\n')
-    try:  # For HPC when device 0 not available. Error: Cuda invalid device ordinal.
-        model.to(device)
-    except RuntimeError:
-        warnings.warn(f"Unable to use device. Trying device 0...\n")
-        device = torch.device(f'cuda:0' if torch.cuda.is_available() and lst_device_ids else 'cpu')
-        model.to(device)
-    model, criterion, optimizer, lr_scheduler = set_hyperparameters(params,
-                                                                    num_classes_corrected,
-                                                                    model,
-                                                                    checkpoint,
-                                                                    dontcare)
-
-    criterion = criterion.to(device)
-
```

Comment:
A lot of code moving around. Have you tested extensively to make sure nothing might break? Is there a general reason for these movements? Getting new features to work, structuring things more logically, etc.?

Reply:
hey! I had the same though again :) any answer @victorlazio109 ?

Reply:
Oh I see! I hadn't seen your previous answer. Thanks.